crates/bevy_ecs/src/query/iter.rs

```diff
@@ -397,11 +378,8 @@ where
     }
 }
 
-impl<'w, 's, Q: WorldQuery, QF, F: WorldQuery> QueryIterationCursor<'w, 's, Q, QF, F>
-where
-    QF: Fetch<'w, State = Q::State>,
-{
-    const IS_DENSE: bool = QF::IS_DENSE && <QueryFetch<'static, F>>::IS_DENSE;
+impl<'w, 's, Q: WorldQuery, F: WorldQuery> QueryIterationCursor<'w, 's, Q, F> {
+    const IS_DENSE: bool = Q::Fetch::IS_DENSE && F::Fetch::IS_DENSE;
```

Comment:
Shouldn't this only rely on whether or not Q is dense? Not your bug though.

Reply:
nope, if you have `Query<&TableComponent, With<SparseComponent>>` its not a dense iterator